scripts/setup_apn_key.sh

```
@@ -0,0 +1,225 @@
+while [[ ! $APN_KEY_ID =~ ^[A-Z0-9]{10}$ ]]; do
+    print_error "Invalid Key ID format. Must be 10 characters (letters and numbers)"
+    read -p "Enter APN Key ID: " APN_KEY_ID
+done
+print_success "APN Key ID: $APN_KEY_ID"
+echo
+
+# Locate APN key file
+print_info "Looking for APN .p8 key file..."
+
+# Find .p8 files
+P8_FILES=($(find . -maxdepth 2 -name "AuthKey_*.p8" 2>/dev/null | grep -v "secrets/AuthKey.p8"))
```

Shellcheck (0.11.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)